CHANGELOG/v1.7.0.md

```diff
@@ -0,0 +1,433 @@
+## 👌 Kubernetes version support
+
+- Management Cluster: v1.26.x -> v1.29.x
+- Workload Cluster: v1.24.x -> v1.29.x
+
+[More information about version support can be found here](https://cluster-api.sigs.k8s.io/reference/versions.html)
+
+
+## Highlights
+
+* Set MachinePool feature flag default to true + Beta
+* Enable kubeadm feature gates mutation
+* Add public function to create new CRD migrator
+* Add MachineSetReady condition to MachineDeployment
+
+## Changes since v1.6.0
+## :chart_with_upwards_trend: Overview
+- 339 new commits merged
+- 5 breaking changes :warning:
+- 12 feature additions ✨
+- 48 bugs fixed 🐛
+
+## :memo: Proposals
+- MachinePool: Update MachinePool Machines proposal with diagrams (#9664)
+
+## :warning: Breaking Changes
+- API: Remove v1alpha4 API Version (#9939)
```

Comment:
This was partially reverted in (below line 49): 
```
- API: Restore v1alpha3/v1alpha4 conversion to fix SSA issue & add e2e test coverage (#10147)
```

However, the restore is a bit different from what we had before, it also restores v1alpha3 for conversion. 

With this release, v1alpha4 types cannot be imported anymore in external go modules (which still can be argued to be a breaking changes)

But @sbueringer may have the best take on this :-)

Reply:
I added a comment about this below